server/src/engine/socket.ts

```diff
@@ -85,60 +84,39 @@ export function isClosed() {
 let lastManualStopTimestamp: number = 0
 
 export function initialiseSocket({ uri, onOpen, onClose }: InitialiseSocketInput) {
-  if (!uri) {
-    throw 'Must be called with an uri'
-  }
-  webSocket = new WebSocket(uri)
+  webSocket = new ReconnectingWebSocket(uri, [], {
+    WebSocket,
+    maxRetries: 5,
+  })
 
-  webSocket.on('open', () => {
+  webSocket.addEventListener('open', () => {
     webSocketOpen = true
-    onOpen && setTimeout(onOpen!, 0)
+    if (onOpen !== undefined) {
+      setTimeout(onOpen, 0)
+    }
   })
 
-  webSocket.on('close', () => {
+  webSocket.addEventListener('close', () => {
     webSocketOpen = false
     if (lastManualStopTimestamp + 15000 < Date.now()) {
-      // This happends when the connections breaks unintentionally
+      // This happens when the connections breaks unintentionally
       console.log(USER_MESSAGE.CONNECTION_LOST())
-      tryToConnect({ uri, onOpen, onClose }, 5).then(connected => {
-        if (!connected) {
-          console.log(USER_MESSAGE.CONNECTION_LOST_RESTARTING())
-          sendNotification(jobs.Request.internalRestart)
-        }
-      })
       return
     }
-    onClose && setTimeout(onClose!, 0)
+
+    if (onClose !== undefined) {
+      setTimeout(onClose, 0)
+    }
   })
 
-  webSocket.on('message', (data: string) => {
-    const flixResponse: FlixResponse = JSON.parse(data)
-    const job: jobs.EnqueuedJob = jobs.getJob(flixResponse.id)
+  webSocket.addEventListener('message', message => {
+    const flixResponse: FlixResponse = JSON.parse(message.data)
+    const job = jobs.getJob(flixResponse.id)
 
     handleResponse(flixResponse, job)
   })
 }
 
-async function tryToConnect({ uri, onOpen, onClose }: InitialiseSocketInput, times: number) {
-  const uriPort = parseInt(uri.slice(-4))
-  getPort({ port: uriPort }, (err, freePort) => {
-    if (uriPort === freePort) {
-      // This happens if the previously used port is now free
-      sendNotification(jobs.Request.internalRestart)
```

Comment:
Do we need jobs.Request.internalRestart anymore?

If so, does it still work correctly?